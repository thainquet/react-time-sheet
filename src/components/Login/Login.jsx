import React from 'reactn';


import { withFormik } from 'formik'
import * as Yup from 'yup'

import Input from '@material-ui/core/Input'
import FormControl from '@material-ui/core/FormControl'
import InputLabel from '@material-ui/core/InputLabel'
import Button from '@material-ui/core/Button'
import Paper from '@material-ui/core/Paper'
import Grid from '@material-ui/core/Grid'
import FormHelperText from '@material-ui/core/FormHelperText'
import Typography from '@material-ui/core/Typography'


const Login = props => {
    const {
        values,
        handleChange,
        errors,
        touched
    } = props;
    console.log(props)
    return (
        <div>
            <Grid container justify='center' alignContent='center'>
                <Grid item xs={6} md={4}>
                    <Paper elevation={4} style={{ padding: '20px 15px', marginTop: '30px' }}>
                        <Typography variant="headline" gutterBottom>
                            Signup
                        </Typography>
                        
                        <FormControl fullWidth margin='normal'>
                            <InputLabel>Username</InputLabel>
                            <Input
                                name='username'
                                value={values.username}
                                onChange={handleChange}
                                autoComplete="off"
                                fullWidth
                            />
                            {touched[values.username] && <FormHelperText>{errors.username}</FormHelperText>}
                        </FormControl>

                        <FormControl fullWidth margin='normal'>
                            <InputLabel>Password</InputLabel>
                            <Input
                                name='password'
                                value={values.password}
                                onChange={handleChange}
                                fullWidth
                            />
                            {touched.password && <FormHelperText>{errors.password}</FormHelperText>}
                        </FormControl>

                        <FormControl fullWidth margin='normal'>
                            <InputLabel>Email</InputLabel>
                            <Input
                                name='email'
                                value={values.email}
                                onChange={handleChange}
                                fullWidth
                            />
                            {touched.email && <FormHelperText>{errors.email}</FormHelperText>}
                        </FormControl>

                        <FormControl fullWidth margin='normal'>
                            <Button
                                color='primary'
                                type='submit'
                            >
                                Signup
                                </Button>
                        </FormControl>
                    </Paper>
                </Grid>
            </Grid>
        </div>
    )
}

const formikForm = withFormik({
    mapPropsToValues() {
        return {
            username: '',
            password: '',
            email: ''
        }
    },
    validationSchema: Yup.object().shape({
        username: Yup.string()
            .required("need")
            .min(6, "6 kt"),
        password: Yup.string()
            .required('Password is required')
            .min(8, 'Password must have min 8 characters'),
        email: Yup.string()
            .required('Email is required')
            .email('Email is invalid')
    })
})(Login);


export default formikForm;

