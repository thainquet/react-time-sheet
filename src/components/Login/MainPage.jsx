import * as React from 'react';

import Form from 'react-bootstrap/Form'
import { Link } from 'react-router-dom'

const MainPage = () => {
  return (
    <div>
      <Form>
        <Form.Group controlId="formBasicEmail">
          <Form.Label>Email address</Form.Label>
          <Form.Control type="email" placeholder="Enter email" />
        </Form.Group>

        <Form.Group controlId="formBasicPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control type="password" placeholder="Password" />
        </Form.Group>
        {/* <Button component={Link} to="/review">Login</Button> */}
        <Link to='/home'>Login</Link>
      </Form>
    </div>
  );
}


export default MainPage;

