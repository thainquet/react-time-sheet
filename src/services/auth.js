import axios from 'axios'

const register = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/register`, params);
}

const login = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/login`, params);
}

const changePassword = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/changePass`, params);
}

const forgotPassword = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/forgotPass`, params);
}

export default {
  login, changePassword, forgotPassword, register
}