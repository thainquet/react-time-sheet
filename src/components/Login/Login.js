
import React, { useState, setGlobal } from 'reactn';
// import { withRouter } from 'react-router'
import { isLogin } from 'helpers/auth'
import { authServices } from 'services'

const Login = (props) => {
  const { initFormParams = { username: '', password: '' } } = props;
  if (isLogin())
    props.history.push('/home')

  const [form, setFormInfo] = useState(initFormParams);

  const [res, resMessage] = useState('')

  const handleLogin = async (event) => {
    if (!form.username || !form.password)
      resMessage(res => res = 'All fields required!')
    else {
      const params = {
        username: form.username,
        password: form.password
      }
      const result = await authServices.login(params)

      resMessage(res => res = result.data.message)

      if (result.data.code === 200) {
        setGlobal({ username: form.username });
        localStorage.setItem('auth', true)
        localStorage.setItem('setupTime', new Date().getTime())
        props.history.push('/home');
      } else {
        localStorage.setItem('auth', false)
      }
    }
  }

  const updateField = event => {
    resMessage(res => res = '')
    setFormInfo({
      ...form,
      [event.target.name]: event.target.value
    });
  };

  const handlePress = event => {
    if (event.key === 'Enter') handleLogin();
  }

  return (
    <>
      <div>
        <label htmlFor="">Username</label>
        <input id="username" type="text" name="username" autoFocus autoComplete="off" value={form.username} onChange={updateField} />
      </div>

      <div>
        <label htmlFor="">Password</label>
        <input id="password" type="password" name="password" autoComplete="off" value={form.password} onChange={updateField} onKeyPress={handlePress} />
        <br />
      </div>

      <pre>{res}</pre>

      <button id="login-btn" onClick={handleLogin} >Dang nhap</button>

      <div>
        <button onClick={() => props.history.push("/register")}>Dang ky</button>
        <button onClick={() => props.history.push("/forgot")}>Quen mat khau</button>
      </div>
    </>
  )
}

export default Login;

