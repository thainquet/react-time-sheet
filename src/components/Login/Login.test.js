import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { authServices } from 'services';
import Login from './Login';

const props = {
  initFormParams: {
    username: 'thainq00',
    password: '1'
  },
  history: []
}

it('should render correctly', () => {
  const wrapper = renderer.create(<Login />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

it('has 2 inputs and 3 buttons', () => {
  const wrapper = mount(<Login />);
  expect(wrapper.find('input')).toHaveLength(2);
  expect(wrapper.find('button')).toHaveLength(3);
  wrapper.unmount();
});

it('Login success - change username, password and submit form', () => {
  const wrapper = mount(<Login {...props} />);
  const loginFn = jest.spyOn(authServices, 'login').mockImplementation(() => Promise.resolve({ data: { code: 200, message: 'Thanh cong' } }));
  wrapper.find('input#username').simulate('change', { target: { value: props.initFormParams.username } })
  wrapper.find('input#password').simulate('change', { target: { value: props.initFormParams.password } })
  wrapper.find('button#login-btn').simulate('click')
  expect(loginFn).toHaveBeenCalledWith(
    props.initFormParams
  );
  wrapper.unmount();
});
