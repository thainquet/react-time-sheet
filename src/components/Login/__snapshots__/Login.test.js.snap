// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <div>
    <label
      htmlFor=""
    >
      Username
    </label>
    <input
      autoComplete="off"
      autoFocus={true}
      id="username"
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>,
  <div>
    <label
      htmlFor=""
    >
      Password
    </label>
    <input
      autoComplete="off"
      id="password"
      name="password"
      onChange={[Function]}
      onKeyPress={[Function]}
      type="password"
      value=""
    />
    <br />
  </div>,
  <pre>
    
  </pre>,
  <button
    id="login-btn"
    onClick={[Function]}
  >
    Dang nhap
  </button>,
  <div>
    <button
      onClick={[Function]}
    >
      Dang ky
    </button>
    <button
      onClick={[Function]}
    >
      Quen mat khau
    </button>
  </div>,
]
`;
